airgun/views/ansible_variable.py

```diff
@@ -0,0 +1,116 @@
+    new_variable = Text("//a[contains(@href, '/ansible/ansible_variables/new')]")
+    total_variables = Text("//span[@class='pagination-pf-items-total']")
+    table = SatTable(
+        './/table',
+        column_widgets={
+            'Name': Text("./a"),
+            'Role': Text("./a"),
+            'Type': Text("./a"),
+            'Imported?': Text("./a"),
+            'Actions': Text(".//a[@data-method='delete']"),
+        },
+    )
+    pagination = Pagination()
+    confirm = Pf4ConfirmationDialog()
+
+    @property
+    def is_displayed(self):
+        return self.title.is_displayed and self.new_variable.is_displayed
+
+
+class MatcherTable(CustomParameter):
+    add_new_value = Text("..//a[contains(text(),'+ Add Matcher')]")
+
+
+class MatcherActions(View):
+    """Interface table has Attribute type column that contains select field and
+    text input field."""
+
+    matcher_key = Select(".//select")
+    matcher_value = TextInput(locator=".//input[@class='matcher_value']")
+
+
+class NewAnsibleVariableView(BaseLoggedInView):
+    """View while creating a new Ansible Variable"""
+
+    breadcrumb = BreadCrumb()
+    key = TextInput(id='ansible_variable_key')
+    description = TextInput(id='ansible_variable_description')
+    ansible_role = FilteredDropdown(id='ansible_variable_ansible_role_id')
+    override = Checkbox(id='ansible_variable_override')
+
+    # Accessing all widgets except the ones above requires that the `override` checkbox is filled
+    parameter_type = SatSelect(id='ansible_variable_parameter_type')
+    default_value = TextInput(id='ansible_variable_default_value')
+    hidden_value = Checkbox(id='ansible_variable_hidden_value')
+    expand_optional_input_validator = Text("//h2[@class='expander collapsed']")
+    required = Checkbox(id='ansible_variable_required')
+    validator_type = SatSelect(id='ansible_variable_validator_type')
+    validator_rule = TextInput(id='ansible_variable_validator_rule')
+    attribute_order = TextInput(id='order')
+    merge_overrides = Checkbox(id='ansible_variable_merge_overrides')
+    merge_default = Checkbox(id='ansible_variable_merge_default')
+    avoid_duplicates = Checkbox(id='ansible_variable_avoid_duplicates')
+    submit = Text('//input[@value="Submit"]')
+    cancel = Text("//a[contains(., text()='Cancel']")
+
+    @View.nested
+    class matcher_section(View):
+        add_matcher = Text("//a[contains(@class, 'add_nested_fields')]")
+        params = MatcherTable(
+            locator=".//table[@class='table white-header']",
+            column_widgets={
+                'Attribute type': MatcherActions(),
+                'Value': TextInput(locator=".//textarea[@id='new_lookup_value_value']"),
+                'Actions': Text(".//a"),
+            },
+        )
+
+        def before_fill(self, values):
+            if not self.params.is_displayed:
+                self.add_matcher.click()
+
+    @property
+    def expand_button(self):
```

Comment:
doesn't look like this is actually used anywhere. Its also slightly risky naming, as it returns an element instance, not a button widget instance.